### ISSUE TITLE
Default values from user-specified Avro schema are ignored

### ISSUE THREAD

**Issue by granturing** [databricks/spark-avro#176]
The current `avroSchema` option is ignoring default values for columns. This is useful when needing to handle schema evolution with older data.

For example, currently, any missing columns simply return `null` vs the specified default:

```
val initial = Seq(("test")).toDF("existing_field")

initial.write.format("com.databricks.spark.avro")
  .mode(SaveMode.Overwrite)
  .save("/temp/default_schema_test.avro")

initial.show()
```

```
+--------------+
|existing_field|
+--------------+
|          test|
+--------------+
```

```
val avroSchema = """
{
 "type": "record",
 "name": "my_schema",
 "fields": [
   {
     "name": "existing_field",
     "type": "string"
   },
   {
     "name": "missing_field",
     "type": "string",
     "default": "hello"
   }
 ]
}
"""

val newVersion = sqlContext.read.format("com.databricks.spark.avro")
  .option("avroSchema", avroSchema)
  .load("/temp/default_schema_test.avro")

newVersion.show()
```

```
+--------------+-------------+
|existing_field|missing_field|
+--------------+-------------+
|          test|         null|
+--------------+-------------+

```

Whereas, the expected output should be:

```
+--------------+-------------+
|existing_field|missing_field|
+--------------+-------------+
|          test|        hello|
+--------------+-------------+
```

**Comment by koertkuipers** [comment]
i believe whats going on is worse than just missing defaults. it seems the spark-avro codebase does not rely on avro for schema resolution at all. this means all features of avro which depend on translating data from writer's schema to reader's schema will not work.

**Comment by itaysk** [comment]
@koertkuipers Thanks this. I've been struggling with an issue for too long before looking for issues and arriving here. Is this why the following doesn't work?
http://stackoverflow.com/questions/40405473/skipping-fields-in-a-record-using-spark-avro
do you know if it's planned to support schema evolution, resolution?

**Comment by JoshRosen** [comment]
This is fixed by #195.
